pgx/src/datum/array.rs

```diff
@@ -75,8 +91,8 @@ impl<'a, T: FromDatum> Array<'a, T> {
     /// - `nulls` is non-null
     /// - both `elements` and `nulls` point to a slice of equal-or-greater length than `nelems`
     unsafe fn from_pg(
-        _ptr: *mut pg_sys::varlena,
-        array_type: *mut pg_sys::ArrayType,
+        _ptr: Option<NonNull<pg_sys::varlena>>,
+        array_type: Option<NonNull<pg_sys::ArrayType>>,
         elements: *mut pg_sys::Datum,
         nulls: *mut bool,
```

Comment:
These are being kept as-is for the moment because the actual fix for them is using the new `RawArray` abstraction plus `bitvec` to handle things, removing usage of `deconstruct_array`, and that's a more complex followup.